app/src/main/java/org/ometa/lovemonster/service/LoveMonsterClient.java

```diff
@@ -0,0 +1,204 @@
+     *
+     * @return
+     *      the singleton {@code LoveMonsterClient}
+     */
+    public static LoveMonsterClient getInstance() {
+        return singletonInstance;
+    }
+
+    /**
+     * The parser used to convert responses from the server into model objects.
+     */
+    @NonNull
+    private final ResponseParser responseParser;
+
+    /**
+     * The client used to make asynchronous http requests.
+     */
+    @NonNull
+    private final AsyncHttpClient asyncHttpClient;
+
+    /**
+     * Private constructor used to implement the singleton. Purposely not made protected to avoid
+     * breaking the singleton.
+     * Defaults to use newly instantiated {@link ResponseParser} and {@link AsyncHttpClient} objects.
+     */
+    private LoveMonsterClient() {
+            this(new ResponseParser(), new AsyncHttpClient());
+    }
+
+    /**
+     * Protected constructor used to create an instance. Is protected scope to allow overriding and
+     * easier unit testing.
+     *
+     * @param responseParser
+     *      the response parser to use to parse requests
+     * @param asyncHttpClient
+     *      the http client used to make requests
+     * @throws IllegalArgumentException
+     *      if {@code responseParser} or {@code asyncHttpClient} are {@code null}
+     */
+    protected LoveMonsterClient(@NonNull final ResponseParser responseParser, @NonNull final AsyncHttpClient asyncHttpClient) throws IllegalArgumentException {
+        if (responseParser == null) {
+            throw new IllegalArgumentException("argument `responseParser` cannot be null");
+        }
+        if (asyncHttpClient == null) {
+            throw new IllegalArgumentException("argument `asyncHttpClient` cannot be null");
+        }
+
+        this.responseParser = responseParser;
+        this.asyncHttpClient = asyncHttpClient;
+    }
+
+    /**
+     * Retrieves recent loves asynchronously. Takes a {@link LoveListResponseHandler} which will be
+     * invoked on response completion.
+     *
+     * @param loveListResponseHandler
+     *      the response handler to use on response completion
+     * @param page
+     *      the page of results to send
+     */
+    public void retrieveRecentLoves(@NonNull final LoveListResponseHandler loveListResponseHandler, final int page) {
+        retrieveRecentLoves(loveListResponseHandler, page, -1);
+    }
+
+    /**
+     * Retrieves recent loves asynchronously. Takes a {@link LoveListResponseHandler} which will be
+     * invoked on response completion.
+     *
+     * @param loveListResponseHandler
+     *      the response handler to use on response completion
+     * @param page
+     *      the page of results to send
+     * @param userId
+     *      the id of the user to filter results on. passing any non-positive number will cause this value to be ignored
+     */
+    public void retrieveRecentLoves(@NonNull final LoveListResponseHandler loveListResponseHandler, final int page, final int userId) {
```

Comment:
Should we add a convenience method here so that the caller doesn't need to worry about passing a negative userId?

```
public void retrieveRecentLoves(handler, page) {
  retrieveRecentLoves(handler, page, -1);
}
```